src/lean_spec/subspecs/forkchoice/store.py

```diff
@@ -420,25 +485,27 @@ def produce_block(self, slot: Slot, validator_index: ValidatorIndex) -> Block:
             post_state = advanced_state.process_block(candidate_block)
 
             # Find new valid attestations matching post-state justification
-            new_attestations: list[SignedVote] = []
-            for validator_id, checkpoint in self.latest_known_votes.items():
+            new_attestations: list[ValidatorAttestation] = []
+            for signed in self.latest_known_votes.values():
                 # Skip if target block is unknown in our store
-                if checkpoint.root not in self.blocks:
+                data = signed.message.data
+                if data.target.root not in self.blocks:
                     continue
 
                 # Create attestation with post-state's latest justified as source
-                vote = Vote(
-                    validator_id=validator_id,
-                    slot=checkpoint.slot,
-                    head=checkpoint,
-                    target=checkpoint,
+                attestation_data = AttestationData(
```

Comment:
we need to add a check here if data.source is same as post_state.latest_justified else we should skip adding this attestation data  to the block